lib/spack/spack/environment/environment.py

```diff
@@ -928,6 +932,42 @@ def included_config_scopes(self):
             # allow paths to contain spack config/environment variables, etc.
             config_path = substitute_path_variables(config_path)
 
+            # strip file URL prefix, if needed, to avoid unnecessary remote
+            # config processing for local files
+            config_path = config_path.replace("file://", "")
+
+            if not os.path.exists(config_path):
+                # Stage any remote configuration file(s)
+                if spack.util.url.is_url_format(config_path):
+                    staged_configs = (
+                        os.listdir(self.config_stage_dir)
+                        if os.path.exists(self.config_stage_dir)
+                        else []
+                    )
+                    basename = os.path.basename(config_path)
+                    if basename in staged_configs:
```

Comment:
Are you suggesting a directory hierarchy for the staged files?

Reply:
Sorry no I didn't mean to imply any particular solution. If we wanted to change this, then one possibility is that you could create a new filename that is a hash of the URL path (we do similar things e.g. for package source downloads to avoid collisions.

To be clear though changing the behavior is unnecessary here: my original request is just that this is clarified in the documentation about environment includes.

Reply:
Modified the comment.